internal/agent/device/lifecycle/manager.go

```diff
@@ -92,15 +99,68 @@ func (m *LifecycleManager) Initialize(ctx context.Context, status *v1alpha1.Devi
 }
 
 func (m *LifecycleManager) Sync(ctx context.Context, current, desired *v1alpha1.RenderedDeviceSpec) error {
-	// TODO: implement
+	// this controller currently does not implement a sync operation
 	return nil
 }
 
 func (m *LifecycleManager) AfterUpdate(ctx context.Context, current, desired *v1alpha1.RenderedDeviceSpec) error {
-	// TODO: implement
+	if current.Decommission == nil && desired.Decommission != nil {
+		m.log.Warn("Detected decommissioning request from flightctl service")
+		m.updateWithStartedCondition(ctx)
+		m.log.Warn("Updated Condition to decommissioning started")
+
+		// todo - switch handling based on decommission target type once we support more types
+
+		m.updateWithCompletedCondition(ctx)
+		m.log.Warn("No errors during decommissioning, updated Condition to decommissioning completed")
```

Comment:
is this true what if wipeAndReboot fails

Reply:
`wipeAndReboot` is only called after this point. But, actually, `wipeAndReboot` needs to be called (but `updateWithCompletedCondition` should *not* be) even if the previous handling steps resulted in an error. so, I'm going to add back the `errored` bool.